include/seqan3/alignment/matrix/detail/two_dimensional_matrix.hpp

```diff
@@ -117,7 +117,7 @@ class two_dimensional_matrix
      */
     template <std::ranges::forward_range entries_t>
     //!\cond
-        requires (!std::same_as<entries_t, storage_type>) && (std::convertible_to<value_type_t<entries_t>, value_type>)
+        requires (std::convertible_to<value_type_t<entries_t>, value_type>)
     //!\endcond
     two_dimensional_matrix(number_rows const row_dim, number_cols const col_dim, entries_t entries) :
```

Comment:
why is entries not const?

Reply:
All it's elements get moved and modified. I guess this should be non-const.